src/macros.jl

```diff
@@ -434,7 +434,9 @@ function _constraint_macro(
     # Strategy: build up the code for add_constraint, and if needed we will wrap
     # in a function returning `ConstraintRef`s and give it to `Containers.container`.
     idxvars, indices = Containers._build_ref_sets(_error, c)
-
+    if args[1] in idxvars
+        _error("Index is the same name as the model.")
```

Comment:
I would be more explicit, e.g. One index has the same name `...` than the JuMP model